packages/ra-core/src/controller/useSelectionState.ts

```diff
@@ -1,75 +0,0 @@
-import { useCallback, useEffect, useRef } from 'react';
```

Comment:
I removed this hook because it couldn't work. 

We used it to "manage" selection in Reference inputs & fields (or at least, to provide selection state and modifiers). No child actually used the selection in these components, so we didn't see that it was buggy.

The problem comes from e.g. bulk actions. Let's imagine that a post detail view renders a ReferenceManyField for comments, which displays a datagrid. The user selects a comment record, then clicks on the bulk delete button. The side effects of the BulkDeleteButton call `useUnselectAll`, but since the selection state was local, the rows were never selected. Therefore, after deletion, gthe bulk action toolbar was still visible, and impossible to remove. 

So again, having a non-global store for selected ids for refrence was not a viable option. As we used Redux for selected ids in the listController, I reused the same approach in Reference fields.